sympy/physics/continuum_mechanics/beam.py

```diff
@@ -180,7 +180,16 @@ def __init__(self, length, elastic_modulus, second_moment, area=Symbol('A'), var
             location of the moving load in ILD calculations. By default, it
             is set to ``Symbol('a')``.
         """
-        self.length = length
+        self._user_map = {}
+        self._user_map_inv = {}
+        def _wrap_positive(S):
+            if isinstance(S, Symbol) and not S.is_positive:
+                S_pos = Symbol(str(S) + ' ', positive=True)
+                self._user_map[S] = S_pos
+                self._user_map_inv[S_pos] = S
+                return S_pos
+            return S
```

Comment:
This sort of thing is usually a bad idea and will lead to problems down the line.

Reply:
I agree, we should let the user set assumptions if they desire them, but not set them internally in classes.